### PR TITLE
[JW8-5260] Fix tab-nabbing vulnerability by using noreferrer rel

### PR DESCRIPTION
**Pull request by zetagame**
### This PR will...
Implement a new "open" util function that allows custom configs.

### Why is this Pull Request needed?
Allows a cross-platform implementation equivelant to window.open, but with support for noreferrer, as well as any additional options we may want to add in the future.

#### Addresses Issue(s):

JW8-5260

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🛠 There are modified src files, but no test changes. Add tests if you're able to.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  🗿 Set a milestone. It should be the ticket's fix version in JIRA.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/js/utils/dom.js [hunk 1]
```diff
@@ -209,3 +209,11 @@ export function nextSibling(element) {
 export function previousSibling(element) {
     return element.previousElementSibling;
 }
+
+export function open(link, target, additionalOptions = {}) {
+    let a = document.createElement('a');
```
Comment (jnatalzia) [inline src/js/utils/dom.js]: let's write some tests!

File: src/js/utils/dom.js [hunk 2]
```diff
@@ -209,3 +209,11 @@ export function nextSibling(element) {
...
+    a.href = link;
+    a.target = target;
+    a = Object.assign(a, additionalOptions);
+    a.click();
```
Comment (jnatalzia) [inline src/js/utils/dom.js]: are there any potential browser security features that would block this click vs. a `window.open`?

File: src/js/utils/dom.js (hunk 1 above)
Comment (jnatalzia) [reply src/js/utils/dom.js]: you should be able to mock `document.createElement` temporarily for the purpose of the test so you can assert the result of these changes.

File: src/js/utils/dom.js [hunk 3]
```diff
@@ -209,3 +209,11 @@ export function nextSibling(element) {
 export function previousSibling(element) {
     return element.previousElementSibling;
 }
+
+export function open(link, target, additionalOptions = {}) {
```
Comment (jnatalzia) [inline src/js/utils/dom.js]: this is super pedantic but `openLink` might be more descriptive here - less potential for variable collisions and whatnot too

File: src/js/utils/dom.js (hunk 2 above)
Comment (zetagame) [reply src/js/utils/dom.js]: Nope! The only issues that prevent click are on really old versions of safari we don't officially support:

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click

Comment (jnatalzia) [reply src/js/utils/dom.js]: 👍

File: test/unit/dom-test.js [hunk 4]
```diff
@@ -393,4 +394,28 @@ describe('dom', function() {
         expect('bounds should not be empty when element has layout').to.not.equal(bounds(element), emptyBound);
     });
 
+    it.only('opens a link', function() {
```
Comment (jnatalzia) [inline test/unit/dom-test.js]: remove the `only` here

**Pull request merged**